src/presenter/task.js

```diff
@@ -0,0 +1,129 @@
+import TaskView from '../view/task.js';
+import TaskEditView from '../view/task-edit.js';
+import {
+  renderLastPlaceElement,
+  replaceDOMElement,
+  removeElement
+} from '../utils/render.js';
+
+const Mode = {
+  DEFAULT: `DEFAULT`,
+  EDIT: `EDIT`
+};
+
+export default class TaskPresenter {
+  constructor(taskListContainer, changeData, changeMode) {
+    this._taskListContainer = taskListContainer;
+    this._changeData = changeData;
+    this._changeMode = changeMode;
+    this._mode = Mode.DEFAULT;
+    this._task = null;
+
+    this._taskComponent = null;
+    this._taskFormComponent = null;
+
+    // handlers
+    this._archiveClickHandler = this._archiveClickHandler.bind(this);
+    this._editClickHandler = this._editClickHandler.bind(this);
+    this._favoriteClickHandler = this._favoriteClickHandler.bind(this);
+    this._formSubmitClickHandler = this._formSubmitClickHandler.bind(this);
+    this._documentEscapeHandler = this._documentEscapeHandler.bind(this);
+  }
+
+  init(task) {
+    this._task = task;
+
+    const prevTaskComponent = this._taskComponent;
+    const prevTaskFormComponent = this._taskFormComponent;
+
+    this._taskComponent = new TaskView(task);
+    this._taskFormComponent = new TaskEditView(task);
+
+    this._taskComponent.setArchiveClickHandler(this._archiveClickHandler);
+    this._taskComponent.setEditClickHandler(this._editClickHandler);
+    this._taskComponent.setFavoriteClickHandler(this._favoriteClickHandler);
+
+    this._taskFormComponent.setFormSubmitHandler(this._formSubmitClickHandler);
+
+    if (!prevTaskComponent || !prevTaskFormComponent) {
+      renderLastPlaceElement(this._taskListContainer, this._taskComponent);
+      return;
+    }
+
+    if (this._mode === Mode.DEFAULT) {
+      replaceDOMElement(this._taskComponent, prevTaskComponent);
+    }
+
+    if (this._mode === Mode.EDIT) {
+      replaceDOMElement(this._taskFormComponent, prevTaskFormComponent);
+    }
```

Comment:
```suggestion
    if (this._mode === Mode.DEFAULT) {
      replaceDOMElement(this._taskComponent, prevTaskComponent);
    } else if (this._mode === Mode.EDIT) {
      replaceDOMElement(this._taskFormComponent, prevTaskFormComponent);
    }
```
А ещё лучше через switch...case